src/sources/soundsourceffmpeg.h

```diff
@@ -42,7 +42,7 @@ struct ffmpegCacheObject {
 
 class SoundSourceFFmpeg : public SoundSource {
   public:
-    explicit SoundSourceFFmpeg(QUrl url);
+    explicit SoundSourceFFmpeg(const QUrl& url);
     ~SoundSourceFFmpeg();
```

Comment:
AudioSource inherits the virtual destructor from AudioSignal. Only the base 
class needs to declare the destructor as virtual.

The same is true for regular member functions. But if you override a virtual 
member function you should decorate it with the C++11 keyword 'override' to 
reveal your intention and let the compiler check if you actually override a 
virtual function.

On 05/11/2016 08:39 PM, RJ Ryan wrote:
>
> In src/sources/soundsourceffmpeg.h 
> <https://github.com/mixxxdj/mixxx/pull/949#discussion_r62902908>:
>
> > @@ -42,7 +42,7 @@ struct ffmpegCacheObject {
> >
> >  class SoundSourceFFmpeg : public SoundSource {
> >    public:
> > -    explicit SoundSourceFFmpeg(QUrl url);
> > +    explicit SoundSourceFFmpeg(const QUrl& url);
> >      ~SoundSourceFFmpeg();
>
> Unrelated to your PR -- I was about to suggest adding override to 
> ~SoundSourceFFmpeg() but I realized AudioSource doesn't have a virtual 
> destructor. That seems like a problem since we delete pointers to 
> AudioSource via QSharedPointer in CachingReaderWorker and AnalyzerQueue.
>
> As I understand it, this is undefined behavior in C++, right? We have no 
> guarantee ~SoundSourceFFmpeg() will run by the C++ standard.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/949/files/b1d3439def629c17148cc4765f54f191cb4c96c4#r62902908>
>